store/evm_aux/store.go

```diff
@@ -49,51 +71,51 @@ type ChildTxRef struct {
 }
 
 type EvmAuxStore struct {
-	db *leveldb.DB
+	db          dbm.DB
+	store       *atomicKVStore
+	maxReceipts uint64
 }
 
-func NewEvmAuxStore(db *leveldb.DB) *EvmAuxStore {
-	return &EvmAuxStore{db: db}
+func NewEvmAuxStore(db dbm.DB, maxReceipts uint64) *EvmAuxStore {
+	return &EvmAuxStore{
+		db:          db,
+		store:       NewAtomicKVStore(db),
+		maxReceipts: maxReceipts,
+	}
 }
 
-func (s *EvmAuxStore) Close() error {
-	return s.db.Close()
+func (s *EvmAuxStore) Close() {
+	s.db.Close()
 }
 
 func (s *EvmAuxStore) GetBloomFilter(height uint64) []byte {
-	filter, err := s.db.Get(bloomFilterKey(height), nil)
-	if err != nil && err != leveldb.ErrNotFound {
-		panic(err)
-	}
-	if err == leveldb.ErrNotFound {
+	filter := s.db.Get(bloomFilterKey(height))
+	if len(filter) == 0 {
 		return nil
 	}
 	return filter
 }
 
 func (s *EvmAuxStore) GetTxHashList(height uint64) ([][]byte, error) {
-	protHashList, err := s.db.Get(evmTxHashKey(height), nil)
-	if err != nil && err != leveldb.ErrNotFound {
-		return nil, err
-	}
-	if err == leveldb.ErrNotFound {
+	protHashList := s.db.Get(evmTxHashKey(height))
+	if len(protHashList) == 0 {
 		return [][]byte{}, nil
 	}
 	txHashList := types.EthTxHashList{}
-	err = proto.Unmarshal(protHashList, &txHashList)
+	err := proto.Unmarshal(protHashList, &txHashList)
 	return txHashList.EthTxHash, err
 }
 
-func (s *EvmAuxStore) SetBloomFilter(tran *leveldb.Transaction, filter []byte, height uint64) error {
-	return tran.Put(bloomFilterKey(height), filter, nil)
+func (s *EvmAuxStore) SetBloomFilter(filter []byte, height uint64) {
```

Comment:
No need to export this function anymore, it's only called from this package now.

Reply:
Done